src/mon/OSDMonitor.cc

```diff
@@ -2542,17 +2542,23 @@ class C_AckMarkedDown : public C_MonOp {
     MonOpRequestRef op)
     : C_MonOp(op), osdmon(osdmon) {}
 
-  void _finish(int) override {
+  void _finish(int r) override {
     MOSDMarkMeDown *m = static_cast<MOSDMarkMeDown*>(op->get_req());
+    if(-EAGAIN == r) {
+    	osdmon->dispatch(op);
+    } else if(ECANCELED == r) {
+    	return;
```

Comment:
might want to use
```c++
osdmon->mon->no_reply(op);
```
to notify the proxy peon if any?

Reply:
Also, space after if here :)

Reply:
I think we need to deal with the message again. Mon needs to mark OSD down as soon as possible.